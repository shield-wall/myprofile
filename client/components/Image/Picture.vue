<template>
  <img :src="computedSource" :alt="alt" @error="setFallbackImageUrl(type)">
</template>

<script>
export default {
  name: 'Picture',
  props: {
    path: { type: String, required: true },
    alt: { type: String, required: true }
  },
  data () {
    const transloaditEnv = process.env.FILE_PROVIDER.transloadit
    const imageDefaultEnv = process.env.IMAGE.default

    const data = { prefix: null, host: null, providerUrl: null, defaultProfileImage: null, defaultBackgroundImage: null }

    if (transloaditEnv) {
      data.prefix = transloaditEnv.prefix
      data.host = transloaditEnv.host
      data.providerUrl = transloaditEnv.host + '/' + transloaditEnv.prefix
    }

    if (imageDefaultEnv) {
      data.defaultProfileImage = imageDefaultEnv.profile
      data.defaultBackgroundImage = imageDefaultEnv.background
    }

    return data
  },
  computed: {
    computedSource () {
      const env = process.env.NODE_ENV

      if (env === 'development') {
        return require(`~/assets/images/${this.prefix}${this.path}`)
      }

      return this.providerUrl + this.path
    }
  },
  methods: {
    setFallbackImageUrl (type) {
      if (type === 'profile') {
        event.target.src = this.providerUrl + this.defaultProfileImage
        return
      }

      event.target.src = this.providerUrl + this.defaultBackgroundImage
    }
  }
}
</script>
