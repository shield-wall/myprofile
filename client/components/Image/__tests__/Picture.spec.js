import { shallowMount } from '@vue/test-utils'
import Picture from '../Picture'

function factory () {
  return shallowMount(Picture, {
    propsData: {
      path: '/foo.webp',
      alt: 'foo alt text.'
    }
  })
}

describe('Picture component.', function () {
  it('is getting the correct image HTML.', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
