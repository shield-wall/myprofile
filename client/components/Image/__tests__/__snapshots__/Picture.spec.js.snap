// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture component. is getting the correct image HTML. 1`] = `<img src="null/foo.webp" alt="foo alt text.">`;
